### PR TITLE
Fix meta ref generation

### PR DESCRIPTION
**Pull request by davidalk**
This fixes the RUNE ISSUE-812, please see [here](https://github.com/finos/rune-dsl/issues/812) for full details of issue

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend [hunk 1]
```diff
@@ -84,37 +84,43 @@ class MetaFieldGenerator {
...
-			val refs = nsc.value.filter(Data).flatMap[expandedAttributes].filter[hasMetas && metas.exists[name=="reference" || name=="address"]].toSet
-			
-			for (ref:refs) {
-				val targetModel = ref.type.model
-				if (targetModel.shouldGenerate) {
+			val refs = nsc.value.filter(Data).flatMap[expandedAttributes].filter [
+				hasMetas && metas.exists[name == "reference" || name == "address"]
+			].toSet
+
+			if (nsc.value.head.model.shouldGenerate) {
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend:95]: I think we can get rid of these checks altogether. This check is already performed by the caller (`RosettaGenerator`)

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend [hunk 2]
```diff
@@ -84,37 +84,43 @@ class MetaFieldGenerator {
...
-				if (targetModel.shouldGenerate) {
-					val targetPackage = new RootPackage(targetModel)
-					val metaJt = meta.toMetaJavaType
-					
-					if (ctx.cancelIndicator.canceled) {
-						return
-					}
-					fsa.generateFile('''«metaJt.canonicalName.withForwardSlashes».java''', fieldWithMeta(targetPackage, metaJt, meta.rosettaType))
-				}
+			if (nsc.value.head.model.shouldGenerate) {
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend:113]: Idem

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend (hunk 1 above)
Comment (davidalk) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend]: done

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend (hunk 2 above)
Comment (davidalk) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend]: done

**Comment by hugohills-regnosys** [comment]
https://github.com/finos/common-domain-model/issues/3364

**Pull request merged**